File: CS/ProcessDuplicatesTarget/ProductsReport.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace ProcessDuplicatesTarget
{
    public partial class ProductsReport : DevExpress.XtraReports.UI.XtraReport
    {
        public ProductsReport()
        {
            InitializeComponent();
        }
        // Display duplicate values.
        public void NoMerge() {
            this.ShowPreviewDialog();
        }
        // Merge duplicate the XRControl.Tag property's values.
        public void MergeByTag() {
            ExpressionBinding expressionBinding = new ExpressionBinding("BeforePrint", "Tag", "ToStr([SupplierID]) + '_' + ToStr([CategoryID])");
            this.xrTableCell2.ExpressionBindings.Add(expressionBinding);
            this.xrTableCell2.ProcessDuplicatesMode = ProcessDuplicatesMode.Merge;
            this.xrTableCell2.ProcessDuplicatesTarget = DevExpress.XtraReports.UI.ProcessDuplicatesTarget.Tag;
            this.ShowPreviewDialog();
        }
        // Merge duplicate values of a report control's data.
        public void MergeByValue() {
            ExpressionBinding expressionBinding = new ExpressionBinding("BeforePrint", "Text", "[CategoryName]");
            this.xrTableCell2.ExpressionBindings.Add(expressionBinding);
            this.xrTableCell2.ProcessDuplicatesMode = ProcessDuplicatesMode.Merge;
            this.xrTableCell2.ProcessDuplicatesTarget = DevExpress.XtraReports.UI.ProcessDuplicatesTarget.Value;
            this.ShowPreviewDialog();
        }
    }
}

File: VB/ProcessDuplicatesTarget/ProductsReport.vb
Imports System.Drawing
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace ProcessDuplicatesTarget

    Public Partial Class ProductsReport
        Inherits XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        ' Display duplicate values.
        Public Sub NoMerge()
            ShowPreviewDialog()
        End Sub

        ' Merge duplicate values of the XRControl.Tag property.
        Public Sub MergeByTag()
            Dim expressionBinding As ExpressionBinding = New ExpressionBinding("BeforePrint", "Tag", "ToStr([SupplierID]) + '_' + ToStr([CategoryID])")
            xrTableCell2.ExpressionBindings.Add(expressionBinding)
            xrTableCell2.ProcessDuplicatesMode = ProcessDuplicatesMode.Merge
            xrTableCell2.ProcessDuplicatesTarget = DevExpress.XtraReports.UI.ProcessDuplicatesTarget.Tag
            ShowPreviewDialog()
        End Sub

        ' Merge duplicate values of a report control's data.
        Public Sub MergeByValue()
            Dim expressionBinding As ExpressionBinding = New ExpressionBinding("BeforePrint", "Text", "[CategoryName]")
            xrTableCell2.ExpressionBindings.Add(expressionBinding)
            xrTableCell2.ProcessDuplicatesMode = ProcessDuplicatesMode.Merge
            xrTableCell2.ProcessDuplicatesTarget = DevExpress.XtraReports.UI.ProcessDuplicatesTarget.Value
            ShowPreviewDialog()
        End Sub
    End Class
End Namespace
